### PR TITLE
Replace satori/go.uuid with its fork gofrs/uuid

### LINKED ISSUE #1543: Move satori/go.uuid to google/uuid
**Bug description**
`github.com/satori/go.uuid` has the following vulnerability.
```
✗ High severity vulnerability found in github.com/satori/go.uuid
  Description: Insecure Randomness
  Info: https://snyk.io/vuln/SNYK-GOLANG-GITHUBCOMSATORIGOUUID-72488
  Introduced through: github.com/satori/go.uuid@1.2.0
  From: github.com/satori/go.uuid@1.2.0
```

However, this library is no longer maintained satori/go.uuid#115

**Affected product area (please put an X in all that apply)**

- [X] Security
- [X] Test and Release

**Expected behavior**
We are not using any unmaintained dependencies

**Steps to reproduce the bug**
1. Clone repo
2. Download snyk
3. Run `snyk test .` inside repo directory 

**Version** (include the SHA if the version is not obvious)

All versions (?)

**Environment where the bug was observed (cloud, OS, etc)**
N/A

**Relevant Debug Output (Logs, manifests, etc)**

**Solution**

After input from #1544 switching to google/uuid makes most sense

### PR DESCRIPTION
**Pull request by PushkarJ**
### What this PR does / why we need it
satori repo is abandoned. Its fork gofrs is well maintained and has high security fixes

Fixes #1543

### Describe testing done for PR

Ran all relevant make commands to ensure tests passed and build was successful

### Release note

```release-note
Replace satori/go.uuid with gofrs/uuid
```

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

#### Special notes for your reviewer
Replaced all imports of the unmaintained library with the new one.
Also had to replace: uuid.NewV4() -> uuid.Must(uuid.NewV4()), because the new library returned an error along with the uuid.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -43,6 +43,7 @@ require (
 	github.com/go-openapi/swag v0.19.14
 	github.com/go-openapi/validate v0.19.8
 	github.com/gobwas/glob v0.2.3
+	github.com/gofrs/uuid v4.2.0+incompatible
```
Comment (rajathagasthya) [inline go.mod:46]: Could we not use `github.com/google/uuid`? It's already an[ indirect dependency](https://github.com/vmware-tanzu/tanzu-framework/blob/9c8711c8e37964ed69d1a55a7e992e61bc061f44/go.mod#L174) and (afaik) provides an identical API.

Comment (randomvariable) [reply go.mod:46]: +1. As that's what we're using upstream https://github.com/search?q=org%3Akubernetes+github.com%2Fgoogle%2Fuuid&type=code

Comment (PushkarJ) [reply go.mod:46]: Thanks Rajath and Naadir. This was a good shout! Created a new PR that does what you suggested: https://github.com/vmware-tanzu/tanzu-framework/pull/1552

**Comment by PushkarJ** [comment]
Closing in favor of #1552

**Pull request closed without merging**